### PR TITLE
test code refactor for model creation and UT for kmeans 

### PR DESCRIPTION
**Pull request by vikash018**
This PR has following changes:
1) Test code refactor –
    • created separate module (model_utils.py) for dataset setup/model initialization/train and predict models for verticapy and python models
    • code modularization for test_base_model_methods.py model

2) Added unit test for Kmeans model
3) remove old unit test for Kmeans /tests/vModel/test_kmeans.py

### CODE REVIEW

**Comment by oualib** [comment]
@vikash018 please see why your tests are failing

**Comment by vikash018** [comment]
> @vikash018 please see why your tests are failing

It was failing due to schema name missing. I have fixed it.

File: verticapy/tests_new/machine_learning/vertica/model_utils.py [hunk 1]
```diff
@@ -0,0 +1,1558 @@
...
+
+le = LabelEncoder()
+
+if sys.version_info < (3, 12):
+    import tensorflow as tf
+
+
+@pytest.fixture(autouse=True)
+def set_plotting_lib():
+    vp.set_option("plotting_lib", "matplotlib")
```
Comment (mail4umar) [inline verticapy/tests_new/machine_learning/vertica/model_utils.py:55]: you can switch back to default by adding a yeild statement
```
yeild
vp.set_option("plotting_lib", "plotly")
```

File: verticapy/tests_new/machine_learning/vertica/test_base_model_methods.py [hunk 2]
```diff
@@ -1343,144 +1115,244 @@ def test_get_vertica_attributes(
...
+        ),
+        **dict.fromkeys(
+            TIMESERIES_MODELS,
+            """{pred_fun_name}( USING PARAMETERS model_name = '{schema_name}.{model_name}', add_mean = True, npredictions = 10 ) OVER ()""",
+        ),
+    }
+    return deploysql_map.get(model_class, None)
+
+
+pytestmark = pytest.mark.parametrize(
```
Comment (mail4umar) [inline verticapy/tests_new/machine_learning/vertica/test_base_model_methods.py:1274]: @vikash018 where do you actually use this? I am intrigued.

**Review by mail4umar — APPROVED** [review]

File: verticapy/tests_new/machine_learning/vertica/test_base_model_methods.py (hunk 2 above)
Comment (vikash018) [reply verticapy/tests_new/machine_learning/vertica/test_base_model_methods.py:1274]: This declares pytest.mark.parametrize at global level which would be applied to all test functions/methods in this the module.

https://pytest.org/en/7.4.x/reference/reference.html#globalvar-pytestmark

File: verticapy/tests_new/machine_learning/vertica/model_utils.py (hunk 1 above)
Comment (vikash018) [reply verticapy/tests_new/machine_learning/vertica/model_utils.py:55]: Thanks added.

**Pull request merged**